kernel: bump 6.6 to 6.6.29

graysky2 opened the pull request:
Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.6.29

Removed upstreamed:
	generic/backport-6.6/740-v6.9-01-netfilter-flowtable-validate-pppoe-header.patch[1]
	generic/backport-6.6/740-v6.9-02-netfilter-flowtable-incorrect-pppoe-tuple.patch[2]
	generic/backport-6.6/790-29-v6.9-net-dsa-mt7530-fix-improper-frames-on-all-25MHz-and-.patch[3]
	generic/backport-6.6/790-31-v6.10-net-dsa-mt7530-fix-enabling-EEE-on-MT7531-switch-on-.patch[4]
        generic/backport-6.6/790-34-v6.10-net-dsa-mt7530-fix-mirroring-frames-received-on-loca.patch[5]
	generic/backport-6.6/790-35-v6.10-net-dsa-mt7530-fix-port-mirroring-for-MT7988-SoC-swi.patch[6]
	mediatek/patches-6.6/963-net-ethernet-mtk_eth_soc-fix-WED-wifi-reset.patch[7]

Manually rebased:
	generic/backport-6.6/790-23-v6.9-net-dsa-mt7530-get-rid-of-priv-info-cpu_port_config.patch

All other patches automatically rebased.
1. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.29&id=a2471d271042ea18e8a6babc132a8716bb2f08b9
2. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.29&id=4ed82dd368ad883dc4284292937b882f044e625d
3. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.29&id=21b9d89d93422221cdda1b82fd075fa3c94a11d9
4. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.29&id=bd41ee1efd478852a0882ce5f136bc2b5e83eff2
5. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.29&id=d1be3960539249a8690ed09a29d0e3bf34189dd2
6. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.29&id=f8de1b6208bf71bd3102548d33dd8475573ad2ea
7. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.29&id=6855f724f19620c3ddff57c349e0abba797475b1

Build system: x86/64
Build-tested: x86/64/AMD Cezanne, flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3
Run-tested: x86/64/AMD Cezanne, flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3

DragonBluep commented on target/linux/generic/backport-6.6/790-34-v6.10-net-dsa-mt7530-fix-mirroring-frames-received-on-loca.patch:
```diff
@@ -35,31 +35,31 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
- 			   PVC_EG_TAG(MT7530_VLAN_EG_CONSISTENT));
- 	}
+@@ -2485,6 +2485,9 @@ mt7530_setup(struct dsa_switch *ds)
+ 	/* Allow mirroring frames received on the local port (monitor port). */
```
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit?h=v6.6.29&id=d1be3960539249a8690ed09a29d0e3bf34189dd2
upstreamed.

graysky2 replied:
Added, thanks.

graysky2 committed:
kernel: bump 6.6 to 6.6.29

Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.6.29

Removed upstreamed:
	generic/backport-6.6/740-v6.9-01-netfilter-flowtable-validate-pppoe-header.patch[1]
	generic/backport-6.6/740-v6.9-02-netfilter-flowtable-incorrect-pppoe-tuple.patch[2]
	generic/backport-6.6/790-29-v6.9-net-dsa-mt7530-fix-improper-frames-on-all-25MHz-and-.patch[3]
	generic/backport-6.6/790-31-v6.10-net-dsa-mt7530-fix-enabling-EEE-on-MT7531-switch-on-.patch[4]
	generic/backport-6.6/790-34-v6.10-net-dsa-mt7530-fix-mirroring-frames-received-on-loca.patch[5]
	generic/backport-6.6/790-35-v6.10-net-dsa-mt7530-fix-port-mirroring-for-MT7988-SoC-swi.patch[6]
	mediatek/patches-6.6/963-net-ethernet-mtk_eth_soc-fix-WED-wifi-reset.patch[7]

Manually rebased:
	generic/backport-6.6/790-23-v6.9-net-dsa-mt7530-get-rid-of-priv-info-cpu_port_config.patch

All other patches automatically rebased.

1. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.29&id=a2471d271042ea18e8a6babc132a8716bb2f08b9
2. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.29&id=4ed82dd368ad883dc4284292937b882f044e625d
3. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.29&id=21b9d89d93422221cdda1b82fd075fa3c94a11d9
4. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.29&id=bd41ee1efd478852a0882ce5f136bc2b5e83eff2
5. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.29&id=d1be3960539249a8690ed09a29d0e3bf34189dd2
6. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.29&id=f8de1b6208bf71bd3102548d33dd8475573ad2ea
7. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.29&id=6855f724f19620c3ddff57c349e0abba797475b1

Build system: x86/64
Build-tested: x86/64/AMD Cezanne, flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3
Run-tested: x86/64/AMD Cezanne, flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3

Signed-off-by: John Audia <therealgraysky@proton.me>

Borromini commented:
Looks like it needs updating? On latest main:

```
$ ./scripts/getver.sh
r26085-2aa6c875f8
$ git am 15302.patch
Applying: kernel: bump 6.6 to 6.6.29
.git/rebase-apply/patch:258: space before tab in indent.
 	wdma_w32(dev, MTK_WDMA_INT_MASK, 0);
.git/rebase-apply/patch:397: space before tab in indent.
 	wdma_w32(dev, MTK_WDMA_INT_MASK, 0);
.git/rebase-apply/patch:1663: space before tab in indent.
 	if ((val & HWTRAP_XTAL_MASK) == HWTRAP_XTAL_40MHZ)
.git/rebase-apply/patch:1664: space before tab in indent.
 		mt7530_pll_setup(priv);
.git/rebase-apply/patch:1698: space before tab in indent.
 	/* Enable Energy-Efficient Ethernet (EEE) and PHY core PLL, since
error: patch failed: target/linux/ipq40xx/patches-6.6/701-net-dsa-add-out-of-band-tagging-protocol.patch:136
error: target/linux/ipq40xx/patches-6.6/701-net-dsa-add-out-of-band-tagging-protocol.patch: patch does not apply
Patch failed at 0001 kernel: bump 6.6 to 6.6.29
hint: Use 'git am --show-current-patch=diff' to see the failed patch
When you have resolved this problem, run "git am --continue".
If you prefer to skip this patch, run "git am --skip" instead.
To restore the original branch and stop patching, run "git am --abort".
```

robimarko commented:
Yeah, it need a rebase and refresh due to recent changes

graysky2 commented:
update in progress

DragonBluep commented:
bcm4908 target kernel config file still needs a refresh. If you don't want to build the toolchain, adding `SPI_BCMBCA_HSSPI=y` should work.

BTW, it seems that the 6.6 kernel is not ready for this target, ref: https://github.com/openwrt/openwrt/commit/c3dc52fb3ce29c4093b83e950b1f8ee579187216. So I think the testing kernel should be disabled for a while.

```
  Altera SPI Controller platform driver (SPI_ALTERA) [N/m/y/?] n
  Analog Devices AXI SPI Engine controller (SPI_AXI_SPI_ENGINE) [N/m/y/?] n
  Broadcom BCM63XX HS SPI controller driver (SPI_BCM63XX_HSSPI) [N/m/y/?] n
  Broadcom BSPI and MSPI controller support (SPI_BCM_QSPI) [N/m/y/?] n
  Broadcom BCMBCA HS SPI controller driver (SPI_BCMBCA_HSSPI) [N/m/y/?] (NEW) make[7]: *** [scripts/kconfig/Makefile:77: syncconfig] Error 1
```

hauke commented:
I removed the patch disabling tetsing kernel for bcm4908 again, it is unrelated to the change and I think the compile problem was not introduced here.

graysky2 commented:
Build system: x86_64
Build-tested: bcm2712
Run-tested: bcm2712/RPi5

```
# uname -a
Linux workbench-openwrt 6.6.34 #0 SMP Wed Jun 19 15:22:23 2024 aarch64 GNU/Linux
```
